File: app/src/main/java/com/evaluation/fragment/MainFragment.java
package com.evaluation.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.evaluation.adapter.CustomListAdapter;
import com.evaluation.dagger.data.DataComponent;
import com.evaluation.model.search.SearchList;
import com.evaluation.navigation.Navigator;
import com.evaluation.network.RestAdapter;
import com.evaluation.retrofitkt.MainActivity;
import com.evaluation.retrofitkt.R;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.reactivex.SingleObserver;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.schedulers.Schedulers;

/**
 * @author Vladyslav Havrylenko
 * @since 09.03.2020
 */
public class MainFragment extends BaseFragment {
    public static final String TAG = MainFragment.class.getCanonicalName();

    public static MainFragment newInstance() {
        return new MainFragment();
    }

    private MainActivity mActivity;

    protected View mRootView;

    @Inject
    Navigator mNavigator;

    @Inject
    RestAdapter restAdapter;

    @BindView(R.id.recycler_view)
    RecyclerView recycleView;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        DataComponent.Injector.getComponent().inject(this);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        if (mRootView == null) {
            mRootView = inflater.inflate(R.layout.main_layout, container, false);
            ButterKnife.bind(this, mRootView);
            recycleView.setLayoutManager(new LinearLayoutManager(mActivity));
        }
        return mRootView;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        loadAssetList();
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mActivity = (MainActivity) getActivity();
    }

    private void loadAssetList() {
        restAdapter.getRestApiService().getSearchData()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new SingleObserver<SearchList>() {
                    @Override
                    public void onSubscribe(Disposable d) {
                    }

                    @Override
                    public void onSuccess(SearchList searchList) {
                        CustomListAdapter adapter = new CustomListAdapter(
                                mActivity,
                                searchList.getSearchList(),
                                selectedProject -> mNavigator.showDetailFragment(selectedProject.getId())
                        );
                        recycleView.setAdapter(adapter);
                    }

                    @Override
                    public void onError(Throwable e) {

                    }
                });
    }
}

File: app/src/main/java/com/evaluation/fragment/MainFragment.kt
package com.evaluation.fragment

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import butterknife.ButterKnife
import com.evaluation.adapter.CustomListAdapter
import com.evaluation.command.ICommand
import com.evaluation.dagger.data.DataComponent
import com.evaluation.model.search.SearchResult
import com.evaluation.navigation.Navigator
import com.evaluation.network.RestAdapter
import com.evaluation.retrofitkt.R
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import javax.inject.Inject

/**
 * @author Vladyslav Havrylenko
 * @since 09.03.2020
 */
class MainFragment : BaseFragment() {

    override val TAG = MainFragment::class.java.canonicalName

    private var mRootView: View? = null

    @Inject
    lateinit var mNavigator: Navigator

    @Inject
    lateinit var restAdapter: RestAdapter

    @BindView(R.id.recycler_view)
    lateinit var recycleView: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        DataComponent.Injector.component!!.inject(this)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        if (mRootView == null) {
            val view = inflater.inflate(R.layout.main_layout, container, false)
            ButterKnife.bind(this, view)
            mRootView = view
            recycleView.layoutManager = LinearLayoutManager(activity)
            loadAssetList()
        }
        return mRootView
    }

    @SuppressLint("CheckResult")
    private fun loadAssetList() {
        restAdapter.restApiService!!.getSearchData()
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe(
                { searchList ->
                    val adapter = CustomListAdapter(
                        activity,
                        searchList.searchResultList,
                        object : ICommand<SearchResult> {
                            override fun execute(param: SearchResult) {
                                mNavigator.showDetailFragment(param.id)
                            }
                        }
                    )
                    recycleView.adapter = adapter
                },
                {})
    }

    companion object {
        fun newInstance(): MainFragment = MainFragment()
    }
}